gcore/src/stack_buffer.rs

```diff
@@ -0,0 +1,54 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021-2023 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Stack allocations utils.
+
+use alloc::vec;
+
+fn with_byte_array<T, const N: usize>(size: usize, f: impl FnOnce(&mut [u8]) -> T) -> T {
+    let mut buffer = [0u8; N];
+    let sub_buffer = &mut buffer[0..size];
+    f(sub_buffer)
+}
+
+/// Calls function `f` with provided byte buffer allocated on stack.
+/// ### IMPORTANT
+/// If buffer size is too big (currently bigger than 0x4000 bytes),
+/// then allocation will be on heap.
+/// If buffer is small enough to be allocated on stack, then real allocated
+/// buffer size will be `size` aligned to upper power of 2.
+pub fn with_byte_buffer<T>(size: usize, f: impl FnOnce(&mut [u8]) -> T) -> T {
+    match size {
+        size if size <= 0x1 => with_byte_array::<_, 0x1>(size, f),
```

Comment:
Simplify this to just if please
How to ref: https://doc.rust-lang.org/std/primitive.usize.html#method.next_power_of_two

Reply:
[Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=201387c35e4a6203c4fd28c5b9c57929) just for you